src/Util/Message.cc

```diff
@@ -717,6 +665,28 @@ CARTA::RemoteFileRequest Message::RemoteFileRequest(int32_t file_id, const strin
     return message;
 }
 
+void Message::AddDirectory(CARTA::FileListResponse& response, casacore::String& name, int64_t date, int32_t item_count) {
+    auto* directory_info = response.add_subdirectories();
+    directory_info->set_name(name);
+    directory_info->set_date(date);
+    directory_info->set_item_count(item_count);
+}
+
+void Message::AddComputedEntry(CARTA::FileInfoExtended& response, std::string name, const std::string &value) {
+    auto entry = response.add_computed_entries();
+    entry->set_name(name);
+    entry->set_value(value);
+    entry->set_entry_type(CARTA::EntryType::STRING);
+}
+
+void Message::AddComputedEntry(CARTA::FileInfoExtended& response, std::string name, const std::string &value, double numeric_value) {
+    auto entry = response.add_computed_entries();
+    entry->set_name(name);
+    entry->set_value(value);
+    entry->set_entry_type(CARTA::EntryType::INT);
```

Comment:
We can't use `INT` for all numeric values -- we need another parameter for setting an explicit type here (either `CARTA::EntryType::INT` or `CARTA::EntryType::FLOAT`).

As discussed previously we can't easily deduce the type from the numeric value parameter (and also there may be cases where we pass in an integer but actually want the entry to be a float). So it's best to be explicit.

We should also have corresponding `AddHeaderEntry` functions for adding header entries that are not computed.